Fixes #18879 - Add prefix filtering by assigned VLAN Group

pheus opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes #18879 - Add prefix filtering by assigned VLAN Group

#### Overview

This PR enhances NetBox by enabling filtering of prefixes by the assigned VLAN Group. Users can now filter using both quick search (by VLAN Group slug) and FilterForm (by ID).

#### Changes Introduced

- **IPAM FilterSet:** Added VLAN Group filters to support filtering by both ID and slug.
- **IPAM FilterForm:** Introduced a `vlan_group_id` filter for direct filtering by VLAN Group.
- **Testing:** Added tests to validate VLAN Group filtering.

Thank you for reviewing this PR!

pheus committed:
feat(ipam): Add VLAN group filters to IPAM FilterSet

Introduces filters for VLAN groups using both ID and slug fields.

pheus committed:
feat(ipam): Add VLAN group filter in IPAM FilterForm

Introduces a `vlan_group_id` filter to IPAM forms for filtering based on
VLAN groups.

jeremystretch commented on netbox/ipam/tests/test_filtersets.py:
```diff
@@ -850,6 +857,15 @@ def test_site(self):
+        vlan_groups = VLANGroup.objects.all()[:2]
+        params = {'vlan_group_id': [vlan_groups[0].pk, vlan_groups[1].pk]}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+        params = {'vlan_group': [vlan_groups[0].slug]}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
```
These two lines can be removed; the following two lines provide sufficient test coverage.
```suggestion
```

pheus replied:
Thanks, I've changed test.

jeremystretch reviewed (changes requested):
Other than one minor note this looks good to go. Thanks!

pheus committed:
feat(ipam): Add VLAN group filtering to tests

Introduces tests for VLAN group filtering in FilterSets. This ensures
correct validation and behavior when filtering by VLAN group.

jeremystretch reviewed (approved):
Thanks @pheus!
